recipe/migrations/slepc321.yaml

```diff
@@ -1,8 +0,0 @@
-__migrator:
-  build_number: 1
-  commit_message: Rebuild for slepc 3.21
-  kind: version
-  migration_number: 1
-migrator_ts: 1711997957.7520285
-slepc:
-- '3.21'
```

Comment:
You could just add `petsc:` here as well. While the migrations would be redundant that way, it'd have the advantage of not (effectively) restarting the slepc migration. Not sure how far it progressed and whether that's a concern, but just as an idea.

Reply:
it started yesterday and hasn't gotten anywhere. There are only a few packages in it. Most things that depend on slepc also depend on petsc.

Usually we do a single migration for all of the petsc/slepc packages, like #4086